docs/tutorials/quantum_reinforcement_learning.ipynb

```diff
@@ -0,0 +1,1564 @@
+    "        action = int(tf.argmax(q_vals[0]).numpy())\n",
+    "    else:\n",
+    "        action = np.random.choice(n_actions)\n",
+    "\n",
+    "    # Apply sampled action in the environment, receive reward and next state\n",
+    "    next_state, reward, done, _ = env.step(action)\n",
+    "    \n",
+    "    return state_array, action, next_state, reward, done"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "oDiw3iJywpUb"
+   },
+   "source": [
+    "and a function that updates the Q-function using a batch of interactions:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {
+    "id": "RR2DjesVwpUb"
+   },
+   "outputs": [],
+   "source": [
+    "@tf.function\n",
+    "def Q_learning_update(states, rewards, next_states, actions, done, model, gamma, n_actions):\n",
+    "    states = tf.convert_to_tensor(states)\n",
+    "    rewards = tf.convert_to_tensor(rewards)\n",
+    "    next_states = tf.convert_to_tensor(next_states)\n",
+    "    actions = tf.convert_to_tensor(actions)\n",
+    "    done = tf.convert_to_tensor(done)\n",
+    "\n",
+    "    # Compute their target q_values and the masks on sampled actions\n",
+    "    future_rewards = model_target([next_states])\n",
+    "    target_q_values = rewards + (gamma * tf.reduce_max(future_rewards, axis=1)\n",
+    "                                                   * (1.0 - done))\n",
+    "    masks = tf.one_hot(actions, n_actions)\n",
+    "\n",
+    "    # Train the model on the states and target Q-values\n",
+    "    with tf.GradientTape() as tape:\n",
+    "        tape.watch(model.trainable_variables)\n",
+    "        q_values = model([states])\n",
+    "        q_values_masked = tf.reduce_sum(tf.multiply(q_values, masks), axis=1)\n",
+    "        loss = tf.keras.losses.Huber()(target_q_values, q_values_masked)\n",
+    "\n",
+    "    # Backpropagation\n",
+    "    grads = tape.gradient(loss, model.trainable_variables)\n",
+    "    for optimizer, w in zip([optimizer_in, optimizer_var, optimizer_out], [w_in, w_var, w_out]):\n",
+    "        optimizer.apply_gradients([(grads[w], model.trainable_variables[w])])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "tfXHhqaPwpUb"
+   },
+   "source": [
+    "Define the hyperparameters:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {
+    "id": "SQ937aYPwpUc"
+   },
+   "outputs": [],
+   "source": [
+    "gamma = 0.99\n",
+    "n_episodes = 2000\n",
+    "\n",
+    "# Define replay memory\n",
+    "max_memory_length = 10000 # Maximum replay length\n",
+    "replay_memory = deque(maxlen=max_memory_length)\n",
+    "\n",
+    "epsilon = 1.0  # Epsilon greedy parameter\n",
+    "epsilon_min = 0.01  # Minimum epsilon greedy parameter\n",
+    "decay_epsilon = 0.99 # Decay rate of epsilon greedy parameter\n",
+    "batch_size = 16\n",
+    "update_VQC = 10 # Train the model every update_VQC steps\n",
```

Comment:
This is still a mixture of camel case and snake case. Can we update the name to be `steps_per_update` ?